Feature/show animals

Adamik10 opened the pull request:
#### What does this PR do?
The CardGrid.tsx component now receives data from the parent component and uses it to show a grid of cards with animal data.  
PS: only mobile view available thus far

#### What's next?
- create filters based on animal type
- add more stuff into the data and filter based on those too (price, etc.)
- finalize mobile styling
- add media queries for other devices

#### Is this related to processing of personal data?
Nope

#### Questions for the reviewer:
What do you use instead of console.log's to see what's happening in real time with the code.

Commit:
project setup:
- deleted package.lock.json - we're moving forward with yarn over npm
- added functionality to read json files in tsconfig.json for easy imports of json data
- tried to add webpack not knowing that I didn't need to. This should maybe be removed
- made scss files from our css files
- rest is just esLint and prettify on save changes to make our files pretty and consistent

Commit:
started making an interface where cards with animals show

Commit:
started making an interface where cards with animals show

spaceo committed:
Make it possible for animals query to return an empty array.
And fixing animals reolver dependency injection

spaceo committed:
Deleting package lock from app

Commit:
- created a CardGrid component
  - it shows a static card at the moment
  - next step is to fetch data and dynamically show all animals

spaceo committed:
Adding vscode filewatcher settings

Adamik10 committed:
- added react hooks to the project
- dynamically show all animals in CardGrid.tsx based on retrieved data from Dashboard.tsx
- added an Interfaces.tsx that we might not need anyway

spaceo commented on app/.parcel-cache/1e610f85a7531055.txt:
```diff
@@ -0,0 +1,2 @@
+25430441
```
What is this file used for? If we need parcel-cache, I think we should add an entry to .gitignore like:
```
app/.parcel-cache
```

Adamik10 replied:
okay after reading this: https://stackoverflow.com/questions/60695778/what-is-the-cache-folder-in-parcel-bundler-is-it-necessary-to-push-the-cache
- I added it to git ignore

spaceo commented on app/package.json:
```diff
@@ -49,6 +53,10 @@
-    "@graphql-codegen/cli": "^2.6.2"
+    "@graphql-codegen/cli": "^2.6.2",
+    "concurrently": "^7.0.0",
+    "nodemon": "^2.0.15",
```
what is nodemon and parcel used for?

Adamik10 replied:
ufff.. I am actually not sure. I think I might have added nodemon when I was adding parcel, not knowing that we don't need it bc it's already there. I'm testing if removing it breaks anything

Adamik10 replied:
update: nothing broke, so it's out

spaceo commented on app/src/App.tsx:
```diff
@@ -1,21 +1,19 @@
-import Cats from './Cats';
+import React from "react";
+import { QueryClient, QueryClientProvider } from "react-query";
+import Cats from "./Cats";
```
Guess we don't need Cats anymore?

spaceo commented on app/src/App.tsx:
```diff
@@ -1,21 +1,19 @@
+  return (
+    // Provide the client to your App
+    <QueryClientProvider client={queryClient}>
+      {/* <Cats /> */}
```
Away with this. It's ok to put the cats behind us. It was a fun time but we need to face reality.

spaceo commented on app/src/Cats.tsx:
```diff
@@ -4,29 +4,32 @@ import { useCatQuery, CatQuery } from "./graphql/generated";
     cat(id: $id) {
-      type,
-      name,
+      type
```
The entrie component should be deleted, I think.

spaceo commented on app/src/components/Interfaces.tsx:
```diff
@@ -0,0 +1,17 @@
+// this file actually might not be necessary.
+// bc in CardGrid.tsx I had to use AllAnimalsQuery type instead of any of this
```
Or we merge it together with global types?
No matter what I think that files like this should be in the root of /src and not as a part of components.
And as a rule of thumb I think that interfaces/types that only are used in one place does not need to be stored globally but can reside along the code where it is being used.

spaceo commented on app/src/components/Interfaces.tsx:
```diff
@@ -0,0 +1,17 @@
+// but maybe useful for general interfaces in the future?
+
+export interface ApiStatus {
+  status: "error" | "idle" | "loading" | "success";
```
We could also make an enum out of this. Enums are nice.

spaceo commented on app/src/components/dashboard.scss:
```diff
@@ -0,0 +1,34 @@
+// should I be adding new vriables here or upstream in the index.scss so that other
+// components can also access them :?
+$scarlet: #f05020;
```
I think it could make sense to create a global scss file called either base.scss or colors.scss.

Adamik10 replied:
cool. I moved it all to index.scss

spaceo commented on app/src/index.tsx:
```diff
@@ -1,14 +1,14 @@
-import reportWebVitals from './reportWebVitals';
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.scss";
```
What kind of rule is replacing the double quotes with single ones?

Adamik10 replied:
![image](https://user-images.githubusercontent.com/28546954/157468984-41727ccb-9086-4188-84ba-74319ecb0edd.png)

that would be Prettier 
should I set it on double quotes?

spaceo commented on app/src/components/CardGrid.tsx:
```diff
@@ -0,0 +1,44 @@
+  // I need to start using something else over console logs
+  // asked some programmer friends what tools they use
+  // What do you use Mikkel?
+  // console.log(`status: ${status}`);
```
I have nothing against console.log. But it is also possible to setup vs code to use step debugging. Maybe we could look at that together, one of these days?
Anyways, I think we should remove these comments since they don't add value to the code base :)

Adamik10 replied:
Yes I'd like that very much.
And removinization is eminent :)

spaceo commented on app/src/components/CardGrid.tsx:
```diff
@@ -0,0 +1,44 @@
+      {data.animals.map((animal, index) => {
+        return (
+          <div className="card" key={index}>
+            <img className="card__image" src={animal?.img} alt={animal?.name} />
+            <div className="card__gradient"></div>
+            <h2 className="card__headline">{animal?.name}</h2>
+            <div className="card__body">
+              <p className="card__body__type">{animal?.type}</p>
```
I think we kan skip the ? checking in these lines.
The schema tells us that we get an array that is either empty or with the animal type.

Adamik10 replied:
mmm I am actually getting an error if the ? is not there
![image](https://user-images.githubusercontent.com/28546954/157471510-57167067-2fa7-41cd-a5dc-cb0ac54a5f48.png)

spaceo reviewed (changes requested):
Looks great! Nice to see it progessing! 🥇 
I have a few remarks.
I think you could get a more clean row of commits by using rebase --interactive. It is not crucial for now, but when we are going to work on a "real" project, and especially when it is open source that is meant to be used by other developers, we need to be more tight in our commit history. ;)

Adamik10 committed:
- applied pull request comments:
  - updated gitignore
  - removed nodemon
  - removed Cats.tsx and all connected mentiones
  - removed unnecessary comments
  - removed Interfaces.tsx - once we need it, we can make it in the root
  - removed unused imports

Adamik10 committed:
Cleaning up dependencies
- removed parcel from the project
- moved eslint plugin for react hooks to the app dev dependencies
- removed package.json + lock.json + .gitignore from the main folder
